jobs/source/RejectedErrorCode.cpp

```diff
@@ -18,6 +18,8 @@
 #include <aws/crt/StlAllocator.h>
 #include <aws/crt/StringUtils.h>
 
+#include <assert.h>
```

Comment:
Why is this required? Why not just use AWS_ASSERT?

Reply:
When transforming a string -> enum, if we don't recognize the string that the service returned, it does `assert(0); return static_cast<JobStatus>(-1);`

It's not something we want to be fatal in a release product. I asked Jonathan if he wanted the assert removed, or preserved and he wanted it preserved here.

Sorry for merging before I noticed this comment

Reply:
/home/kevin/deviceagentcpp-build/aws-iot-device-sdk-cpp-v2-prefix/src/aws-iot-device-sdk-cpp-v2/discovery/source/DiscoveryClient.cpp:32:47: error: ‘AWS_ASSERT’ was not declared in this scope

Still an issue I believe

Reply:
This ought to work, the include chain is: `DiscoveryClient.cpp` -> `#include <aws/crt/Types.h>` -> `#include <aws/common/common.h>` -> `#include <aws/common/assert.inl>` ->`#define AWS_ASSERT`

Have you updated to [aws-crt-cpp v0.4.0](https://github.com/awslabs/aws-crt-cpp/releases/tag/v0.4.0)?